core/templates/dev/head/domain/collection/CollectionNodeObjectFactorySpec.js

```diff
@@ -41,32 +41,114 @@ describe('Collection node object factory', function() {
   it('should contain initial prerequisite skill ids', function() {
     var collectionNode = _createEmptyCollectionNode(
       'exp_id0', ['pre_skill1', 'pre_skill0'], ['acq_skill1', 'acq_skill0']);
-    expect(collectionNode.getPrerequisiteSkillList().getSkills()).toEqual([
+    expect(collectionNode.getPrerequisiteSkillIds()).toEqual([
       'pre_skill1', 'pre_skill0'
     ]);
   });
 
   it('should contain initial acquired skill ids', function() {
     var collectionNode = _createEmptyCollectionNode(
       'exp_id0', ['pre_skill1', 'pre_skill0'], ['acq_skill1', 'acq_skill0']);
-    expect(collectionNode.getAcquiredSkillList().getSkills()).toEqual([
+    expect(collectionNode.getAcquiredSkillIds()).toEqual([
       'acq_skill1', 'acq_skill0'
     ]);
   });
 
   it('should provide a mutable prerequisite skill list object', function() {
     var collectionNode = _createEmptyCollectionNode('exp_id0');
-    var prerequisiteSkillList = collectionNode.getPrerequisiteSkillList();
-    prerequisiteSkillList.addSkill('pre_skill');
-    expect(collectionNode.getPrerequisiteSkillList().containsSkill(
-      'pre_skill'));
+    var prerequisiteSkillIds = collectionNode.getPrerequisiteSkillIds();
+    prerequisiteSkillIds.push('pre_skill');
+    expect(
+      collectionNode.getPrerequisiteSkillIds().indexOf('pre_skill') !== -1);
   });
 
   it('should provide a mutable acquired skill list object', function() {
     var collectionNode = _createEmptyCollectionNode('exp_id0');
-    var acquiredSkillList = collectionNode.getAcquiredSkillList();
-    acquiredSkillList.addSkill('acq_skill');
-    expect(collectionNode.getAcquiredSkillList().containsSkill('acq_skill'));
+    var acquiredSkillIds = collectionNode.getAcquiredSkillIds();
+    acquiredSkillIds.push('acq_skill');
```

Comment:
I considered doing this too, but I don't think it's necessary. Skill ID validation happens at a different layer of the code, and being able to do custom names here can help make the test clearer.